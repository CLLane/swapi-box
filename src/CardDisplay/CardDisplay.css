.single-card {
  float: right;
  margin-right: 15%;
  margin-top: 50px;
  box-shadow:
    0 0 0.1vw  0.3vw #fff7f7,   
    0 0 0.4vw  0.4vw #FFE81F, 
    0 0   4vw  0.3vw rgb(250, 238, 124),
    inset 0 0 1.5vw  0.3vw rgb(250, 238, 124),
    inset 0 0 0.4vw  0.1vw #FFE81F,
    inset 0 0 0.5vw  0.1vw #fff7f7;
    border-radius: 6px;
  width: 700px;
  height: 500px;
  text-align: center;
  color: white;
  text-decoration: none;
}