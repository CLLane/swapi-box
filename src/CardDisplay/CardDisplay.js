import React from 'react';
import './CardDisplay.css';

const CardDisplay = ({  name, homeworld, species, population, terrain, climate, residents, model, vehicleClass, numberOfPassengers }) => {

  return (
    <div className="single-card">
      {/* <Link to={`${}`} */}
      <section className="big-card">
        <h1>{name}</h1>
        <p>{homeworld || null}</p>
        <p>{species || null}</p>
        <p>{population || null}</p>
        <p>{terrain || null}</p>
        <p>{climate || null}</p>
        <p>{residents || null}</p>
        <p>{model || null}</p>
        <p>{vehicleClass || null}</p>
        <p>{numberOfPassengers || null}</p>
      </section>
    </div>
  );
}

export default CardDisplay;