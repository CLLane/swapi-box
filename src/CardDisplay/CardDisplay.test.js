import React from 'react';
import { shallow } from 'enzyme';
import CardDisplay from './CardDisplay';

describe('CardDisplay', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<CardDisplay
      id={1}
      name='Luke Skywalker'
      homeworld= 'Tatooine'
      species= 'Human'
      population= {200000}
      terrain= 'grasslands, mountains'
      climate= 'temperate'
      model= 'Digger Crawler'
      vehicleClass= 'wheeled'
      numberOfPassengers= '30'
      toggleFav={jest.fn()}
      type = '/people'
      />);

      expect(wrapper).toMatchSnapshot(); 
  })
});