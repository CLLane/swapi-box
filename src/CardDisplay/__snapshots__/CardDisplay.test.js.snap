// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay should match the snapshot with all data passed in correctly 1`] = `
<div
  className="single-card"
>
  <section
    className="big-card"
  >
    <h1
      className="big-card-h1"
    >
      Luke Skywalker
    </h1>
    <h4
      className="big-card-labels"
    >
      Homeworld: 
      <p
        className="big-card-values"
      >
        Tatooine
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Species: 
      <p
        className="big-card-values"
      >
        Human
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Population: 
      <p
        className="big-card-values"
      >
        200000
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Terrain: 
      <p
        className="big-card-values"
      >
        grasslands, mountains
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Climate: 
      <p
        className="big-card-values"
      >
        temperate
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Residents: 
      <p
        className="big-card-values"
      />
    </h4>
    <h4
      className="big-card-labels"
    >
      Model: 
      <p
        className="big-card-values"
      >
        Digger Crawler
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Vehicle Class: 
      <p
        className="big-card-values"
      >
        wheeled
      </p>
    </h4>
    <h4
      className="big-card-labels"
    >
      Number of Passengers: 
      <p
        className="big-card-values"
      >
        30
      </p>
    </h4>
  </section>
</div>
`;
